From what I can see in the logs, there's a problem in `fonttools_script.py`:
```sh
       >   File "/private/tmp/nix-build-notonoto-0.0.3.drv-0/source/fonttools_script.py", line 10, in <module>
       >     from fontTools import merge, ttLib, ttx
       > ModuleNotFoundError: No module named 'fontTools'
```

There's also a problem with `fontmake`:
```sh
       > =========================== short test summary info ============================
       > FAILED tests/test_main.py::test_main_designspace_v5_builds_STAT - assert ['<Version va...wght"/>', ...] == ['<Version va...wght"/>', ...]
       > ============= 1 failed, 78 passed, 6 skipped, 4 warnings in 5.39s ==============
       For full logs, run 'nix log /nix/store/6f2dg18swfj5h8ps996h6ai6v586xbxd-python3.12-fontmake-3.9.0.drv'.
```

I'm investigating the `fontmake` problem.

--- a/pkgs/by-name/no/notonoto-hs/package.nix
+++ b/pkgs/by-name/no/notonoto-hs/package.nix
@@ -1,0 +1,58 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenvNoCC,
+  fontforge,
+  python3,
+}:
+
+let
+  python3' = python3.withPackages (
+    ps: with ps; [
+      fonttools
+      ttfautohint-py
+    ]
+  );
+in
+
+stdenvNoCC.mkDerivation rec {
+  pname = "notonoto-hs";
+  version = "0.0.3";
+
+  src = fetchFromGitHub {
+    owner = "yuru7";
+    repo = "NOTONOTO";
+    tag = "v${version}";
+    hash = "sha256-1dbx4yC8gL41OEAE/LNDyoDb4xhAwV5h8oRmdlPULUo=";
+  };
+
+  nativeBuildInputs = [
+    fontforge
+    python3'
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    fontforge --script fontforge_script.py --hidden-zenkaku-space
+    python3 ./fonttools_script.py
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm444 build/*.ttf -t $out/share/fonts/truetype/notonoto-hs
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Programming font that combines Noto Sans Mono and Noto Sans JP";
+    homepage = "https://github.com/yuru7/NOTONOTO";
+    license = lib.licenses.ofl;
+    maintainers = with lib.maintainers; [ genga898 ];
+    platforms = lib.platforms.all;
+  };
+}

--- a/pkgs/by-name/no/notonoto/package.nix
+++ b/pkgs/by-name/no/notonoto/package.nix
@@ -2,7 +2,19 @@
   lib,
   fetchFromGitHub,
   stdenvNoCC,
+  fontforge,
+  python3,
 }:
+
+let
+  python3' = python3.withPackages (
+    ps: with ps; [
+      fonttools
+      ttfautohint-py
+    ]
+  );
+in
+
 stdenvNoCC.mkDerivation rec {
   pname = "notonoto";
   version = "0.0.3";
@@ -14,10 +26,24 @@ stdenvNoCC.mkDerivation rec {
     hash = "sha256-1dbx4yC8gL41OEAE/LNDyoDb4xhAwV5h8oRmdlPULUo=";
   };
 
+  nativeBuildInputs = [
+    fontforge
+    python3'
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    fontforge --script fontforge_script.py
+    python3 ./fonttools_script.py
+
+    runHook postBuild
+  '';
+
   installPhase = ''
     runHook preInstall
 
-    find . -name '*.ttf' -exec install -m444 -Dt $out/share/fonts/notonoto {} \;
+    install -Dm444 build/*.ttf -t $out/share/fonts/truetype/notonoto
 
     runHook postInstall
   '';
@@ -27,7 +53,6 @@ stdenvNoCC.mkDerivation rec {
     homepage = "https://github.com/yuru7/NOTONOTO";
     license = lib.licenses.ofl;
     maintainers = with lib.maintainers; [ genga898 ];
-    mainProgram = "notonoto";
+    platforms = lib.platforms.all;
   };
-
 }

